Mathlib/Algebra/CharP/LinearMaps.lean

```diff
@@ -0,0 +1,56 @@
+/-
+Copyright (c) 2024 Wanyi He. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Wanyi He, Huanyu Zheng
+-/
+import Mathlib.Algebra.CharP.Algebra
+/-!
+# Characteristic of the ring of linear Maps
+
+This file contains properties of the characteristic of the ring of linear maps.
+The characteristic of the ring of linear maps is determined by its base ring.
+
+## Main Results
+
+- `CharP_if` : For a commutative semiring `R` and a `R`-module `M`,
+  the characteristic of `R` is equal to the characteristic of the `R`-linear
+  endomorphisms of `M` when `M` contains an element `x` such that
+  `r • x = 0` implies `r = 0`.
+
+## Notations
+
+- `R` is a commutative semiring
+- `M` is a `R`-module
+
+## Implementation Notes
+
+One can also deduce similar result via `charP_of_injective_ringHom` and
+  `R → (M →ₗ[R] M) : r ↦ (fun (x : M) ↦ r • x)`. But this will require stronger condition
+  compared to `CharP_if`.
+
+-/
+
+variable {R M : Type*} [CommSemiring R] [AddCommMonoid M] [Module R M]
+
+/-- For a commutative semiring `R` and a `R`-module `M`, if `M` contains an
+  element `x` such that `r • x = 0` implies `r = 0` (finding such element usually
+  depends on specific `•`), then the characteristic of `R` is equal to the
+  characteristic of the `R`-linear endomorphisms of `M`.-/
+theorem CharP_if {p : ℕ} [hchar : CharP R p]
```

Comment:
I'm not sure if this is the right name. Maybe `Module.charP_end` or something similar.

Reply:
Indeed that looks better, implemented, thanks!